### PR TITLE
Fix #307 by accepting avahi null names, and install avahi service file

### LINKED ISSUE #307: libiio only supports Avahi, and doesn't natively support Bonjour on Apple and Windows.
As far as I was able to understand, IIOD can broadcast its availability using Avahi.

Client library libiio can discover such devices on Linux using Avahi library.

Since Avahi and Bonjour are compatible, is there a way to use discovery feature in libiio on macOS somehow or shall this part be implemented separately using macOS native APIs?

### PR DESCRIPTION
**Pull request by rgetz**
This will make sure things like this are functional:

./tests/iio_attr -u ip: -d
IIO context has 5 devices:
	iio:device0: adm1177, found 0 device attributes
	iio:device1: ad9361-phy, found 18 device attributes
	iio:device2: xadc, found 1 device attributes
	iio:device3: cf-ad9361-dds-core-lpc, found 0 device attributes
	iio:device4: cf-ad9361-lpc, found 0 device attributes

Signed-off-by: Robin Getz <robin.getz@analog.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -218,6 +218,7 @@ endif()
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
 if(WITH_NETWORK_BACKEND)
+	message(STATUS "Buidling with Network back end support")
```
Comment (dbogdan) [inline CMakeLists.txt]: Buidling - typo.

Comment (rgetz) [reply CMakeLists.txt]: fixed - sorry

**Comment by rgetz** [comment]
fixed the codacy issue.

**Comment by rgetz** [comment]
the travis-ci push fail was some weird homebrew failure; everything else is OK.

**Pull request merged**